app/services/idv/analytics_events_enhancer.rb

```diff
@@ -1,56 +1,58 @@
-      idv_cancellation_confirmed
-      idv_cancellation_go_back
-      idv_cancellation_visited
-      idv_forgot_password
-      idv_forgot_password_confirmed
-      idv_final
-      idv_gpo_address_letter_enqueued
-      idv_gpo_address_letter_requested
-      idv_in_person_ready_to_verify_visit
-      idv_letter_enqueued_visit
-      idv_personal_key_acknowledgment_toggled
-      idv_personal_key_downloaded
-      idv_personal_key_submitted
-      idv_personal_key_visited
-      idv_phone_confirmation_form_submitted
-      idv_phone_confirmation_otp_rate_limit_attempts
-      idv_phone_confirmation_otp_rate_limit_locked_out
-      idv_phone_confirmation_otp_rate_limit_sends
-      idv_phone_confirmation_otp_resent
-      idv_phone_confirmation_otp_sent
-      idv_phone_confirmation_otp_submitted
-      idv_phone_confirmation_otp_visit
-      idv_phone_confirmation_vendor_submitted
-      idv_phone_error_visited
-      idv_phone_of_record_visited
-      idv_phone_otp_delivery_selection_visit
-      idv_phone_otp_delivery_selection_submitted
-      idv_proofing_resolution_result_missing
-      idv_enter_password_submitted
-      idv_enter_password_visited
-      idv_please_call_visited
-      idv_start_over
+    IGNORED_METHODS = %i[
+      idv_acuant_sdk_loaded
+      idv_back_image_added
+      idv_back_image_clicked
+      idv_barcode_warning_continue_clicked
+      idv_barcode_warning_retake_photos_clicked
+      idv_capture_troubleshooting_dismissed
+      idv_consent_checkbox_toggled
+      idv_exit_optional_questions
+      idv_front_image_added
+      idv_front_image_clicked
+      idv_image_capture_failed
+      idv_in_person_ready_to_verify_sp_link_clicked
+      idv_in_person_ready_to_verify_what_to_bring_link_clicked
+      idv_in_person_switch_back_submitted
+      idv_in_person_switch_back_visited
+      idv_link_sent_capture_doc_polling_complete
+      idv_link_sent_capture_doc_polling_started
+      idv_mobile_device_and_camera_check
+      idv_native_camera_forced
+      idv_verify_in_person_troubleshooting_option_clicked
+      idv_warning_action_triggered
+      idv_warning_shown
     ].freeze
 
-    DECORATED_METHODS.each do |method_name|
-      define_method(method_name) do |**kwargs|
-        super(**kwargs, **common_analytics_attributes)
+    def self.included(_mod)
+      raise 'this mixin is intended to be prepended, not included'
+    end
+
+    def self.prepended(mod)
+      mod.instance_methods.each do |method_name|
+        if should_enhance_method?(method_name)
+          mod.define_method method_name do |**kwargs|
+            super(**kwargs, **common_analytics_attributes)
+          end
+        end
       end
     end
 
-    def self.included(_mod)
-      raise 'this mixin is intended to be prepended, not included'
+    def self.should_enhance_method?(method_name)
+      return false unless /^idv_/.match?(method_name)
+
+      return false if IGNORED_METHODS.include?(method_name)
+
+      true
     end
```

Comment:
personally I would add the ignore list first, and then maybe flip the rest around for readability?

```suggestion
    def self.should_enhance_method?(method_name)
      return false if IGNORED_METHODS.include?(method_name)

      /^idv_/.match?(method_name)
    end
```

Could even further micro-optimize a regex out of there by doing a string [`#start_with?`](https://docs.ruby-lang.org/en/3.2/Symbol.html#method-i-start_with-3F)

```suggestion
    def self.should_enhance_method?(method_name)
      return false if IGNORED_METHODS.include?(method_name)

      method_name.start_with?('idv_')
    end
```